File: java/parser/src/main/java/gov/lbl/parser/parser/For_equationVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Equation;
import gov.lbl.parser.domain.For_equation;
import gov.lbl.parser.domain.For_indices;

import static java.util.stream.Collectors.toList;

public class For_equationVisitor extends modelicaBaseVisitor<For_equation> {
    @Override
    public For_equation visitFor_equation(modelicaParser.For_equationContext ctx) {
        For_indicesVisitor for_indicesVisitor = new For_indicesVisitor();
        For_indices for_indices = ctx.for_indices() == null? null : ctx.for_indices().accept(for_indicesVisitor);

        EquationVisitor equationVisitor = new EquationVisitor();
        List<Equation> loop_equations = ctx.equation() == null ? null : ctx.equation()
                                                                                .stream()
                                                                                .map(eqn -> eqn.accept(equationVisitor))
                                                                                .collect(toList());
        return new For_equation(for_indices, loop_equations);
    }
}

File: jsParser/parser/For_equationVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const For_equation = require('../domain/For_equation')

const EquationVisitor = require('./EquationVisitor')
const For_indicesVisitor = require('./For_indicesVisitor')

class For_equationVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitFor_equation (ctx) {
    let for_indices = null
    const loop_equations = []

    if (ctx.for_indices()) {
      const for_indicesVisitor = new For_indicesVisitor.For_indicesVisitor()
      for_indices = for_indicesVisitor.visitFor_indices(ctx.for_indices())
    }

    if (ctx.equation()) {
      const equationVisitor = new EquationVisitor.EquationVisitor()
      ctx.equation().forEach(eqn => {
        loop_equations.push(equationVisitor.visitEquation(eqn))
      })
    }

    return new For_equation.For_equation(for_indices, loop_equations)
  }
};

For_equationVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitFor_equation = this.visitFor_equation
exports.For_equationVisitor = For_equationVisitor
